### PR TITLE
docs: update README.md and use "Universal Ctags" as project name

### PR DESCRIPTION
**Pull request by hirooih**
- update README.md and the introduction of index.rst
- use "Universal Ctags" and "Exuberant Ctags" consistently
- add cress reference links
- fix cress reference links, typos, and markups

### CODE REVIEW

File: docs/news.rst
Comment (masatake) [inline docs/news.rst:701]: Yes, you must use --_xformat.

File: docs/contributions.rst
Comment (masatake) [inline docs/contributions.rst:85]: I wrote the concepts of kinds and roles in ctags(1).

File: docs/contributions.rst
Comment (masatake) [inline docs/contributions.rst:353]: LGTM is needed (1) when the change you made is for files other than you maintain and (2) when the change is not obivous.

Fixing a typo is so obvious, so you don't have to wait LGTM response.

File: docs/news.rst
Comment (masatake) [inline docs/news.rst:228]: yes.

File: docs/news.rst
Comment (masatake) [inline docs/news.rst:355]: yes.

File: docs/contributions.rst
Comment (masatake) [inline docs/contributions.rst:227]: This is useful information.

File: README.md
Comment (masatake) [inline README.md:18]: In the e-ctags project, there were some important developers. So I would like to say
"[Exuberant Ctags](http://ctags.sourceforge.net/) (e-ctags) maintained by Darren Hiebert, ..."

File: README.md
Comment (masatake) [inline README.md:21]: I introduced the term `optlib`. So using `optlib` here can make people confused.

File: README.md
Comment (masatake) [inline README.md:12]: I'm insisting on the word "maintained".
How about " ... is a *matained* implementation of " ?

File: README.md
Comment (masatake) [inline README.md:35]: If we write "maintained" at the top of this doc, we can remove this one.

File: README.md
Comment (masatake) [inline README.md:85]: I'm afraid people using old u-ctags refers the latest documents at the web.
How about following sentences?
Users should first consult the man pages in the source tree (man/ctags.1) or a binary package you installed.
Preformatted manuals for the latest source is at [docs.crags.io](https://docs.ctags.io/en/latest/man-pages.html).

**Comment by masatake** [comment]
Thank you very much.
I put some minor comments. 
I also answered some TODO items with `?`.

File: docs/news.rst
Comment (hirooih) [reply docs/news.rst:687]: I left this after revised the comment.

File: docs/contributions.rst
Comment (hirooih) [reply docs/contributions.rst]: I added the cross reference.

File: docs/contributions.rst
Comment (hirooih) [reply docs/contributions.rst]: Fixed the typo and add description.

File: docs/news.rst
Comment (hirooih) [reply docs/news.rst]: Removed.

File: docs/news.rst
Comment (hirooih) [reply docs/news.rst]: Removed.

File: docs/contributions.rst
Comment (hirooih) [reply docs/contributions.rst]: I can't remember why I thought this section was verbose.  I delete the TODO comment.

File: README.md
Comment (hirooih) [reply README.md]: Fixed

File: README.md
Comment (hirooih) [reply README.md]: Fixed.

File: README.md
Comment (hirooih) [reply README.md]: Fixed.

File: README.md
Comment (hirooih) [reply README.md]: Fixed.

File: README.md
Comment (hirooih) [reply README.md:85]: I did not take this, because It is obvious the information on the web may be newer than one for installed version.
This sentence should be simple to let readers know man pages are primary document.  I had been misunderstanding until I discussed with you about this.

**Comment by hirooih** [comment]
Thank you for your feedbacks.  I fixed them except the last one.

**Review by masatake — APPROVED** [review]

**Pull request merged**
